salt/config/__init__.py

```diff
@@ -2848,14 +2848,14 @@ def is_profile_configured(opts, provider, profile_name, vm_=None):
     # If cloning on Linode, size and image are not necessary.
     # They are obtained from the to-be-cloned VM.
     if driver == 'linode' and profile_key.get('clonefrom', False):
-        non_image_drivers.append('linode')
+        drivers_requiring_an_image.append('linode')
         non_size_drivers.append('linode')
 
     # If cloning on VMware, specifying image is not necessary.
-    if driver == 'vmware' and profile_key.get('image', True):
```

Comment:
The line below will execute:
1. If driver is ``vmware`` and ``image`` is specified in the profile.
2. If driver is ``vmware`` and ``image`` is missing from the profile. 

The line below should execute when ``image`` is not specified in the profile. Change this line to the following:
```
if driver == 'vmware' and 'image' not in list(profile_key.keys()):
```